### PR TITLE
Show a flattened tree in LinkGenerationDecisionTree's DebuggerDisplay…

### LINKED ISSUE #636: Flatten the LinkGenerationDecisionTree to show as debugger display string
Similar in the lines of https://github.com/aspnet/Routing/issues/633, it would be useful to flatten the `LinkGenerationDecisionTree` for easy diagnosis.

### PR DESCRIPTION
**Pull request by kichalla**
…String

[Fixes #636] Flatten the LinkGenerationDecisionTree to show as debugger display string

### CODE REVIEW

**Review by JamesNK — APPROVED** [review]

File: src/Microsoft.AspNetCore.Routing/Internal/LinkGenerationDecisionTree.cs [hunk 1]
```diff
@@ -160,5 +164,51 @@ public int Compare(OutboundMatchResult x, OutboundMatchResult y)
...
         }
+
+        // Example output:
+        //
+        // => action: Buy => controller: Store => version: V1(Matches: Store/Buy/V1)
+        // => action: Buy => controller: Store => version: V2(Matches: Store/Buy/V2)
+        // => action: Buy => controller: Store => area: Admin(Matches: Admin/Store/Buy)
+        // => action: Buy => controller: Products(Matches: Products/Buy)
+        // => action: Cart => controller: Store(Matches: Store/Cart)
+        private string DebuggerDisplayString => ToDebuggerDisplayString();
```
Comment (JamesNK) [inline src/Microsoft.AspNetCore.Routing/Internal/LinkGenerationDecisionTree.cs]: remove

File: src/Microsoft.AspNetCore.Routing/Internal/LinkGenerationDecisionTree.cs [hunk 2]
```diff
@@ -3,12 +3,16 @@
...
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
 using Microsoft.AspNetCore.Routing.DecisionTree;
 using Microsoft.AspNetCore.Routing.Tree;
 
 namespace Microsoft.AspNetCore.Routing.Internal
 {
     // A decision tree that matches link generation entries based on route data.
+    [DebuggerDisplay("{DebuggerDisplayString,nq}")]
```
[the remark this answers is not in the document]
Comment (kichalla) [reply src/Microsoft.AspNetCore.Routing/Internal/LinkGenerationDecisionTree.cs]: When changing it to a 'method' I see that we do not get a way to see the 'view' (zooming thing) of the debugger string,
image
vs.
image

[the remark this answers is not in the document]
Comment (JamesNK) [reply src/Microsoft.AspNetCore.Routing/Internal/LinkGenerationDecisionTree.cs]: Hrm, every other place in routing and MVC uses `DebuggerToString()`

[the remark this answers is not in the document]
Comment (kichalla) [reply src/Microsoft.AspNetCore.Routing/Internal/LinkGenerationDecisionTree.cs]: Ah, seems like there are a mix of these things: https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.RazorPages/PageActionDescriptor.cs#L10

I suppose in other instances where its DebuggerToString, its a single line whereas here the text is broken in multiple lines and so we need the viewer to see it nicely. do you agree?

**Comment by kichalla** [comment]
Ping

File: src/Microsoft.AspNetCore.Routing/Internal/LinkGenerationDecisionTree.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (JamesNK) [reply src/Microsoft.AspNetCore.Routing/Internal/LinkGenerationDecisionTree.cs]: ok, in that case remove `ToDebuggerDisplayString` and just have the property. The method isn't adding anything

**Comment by kichalla** [comment]
https://github.com/aspnet/Routing/commit/6f4c10a664bbb155e5085e822f126feef1d33eb4

**Pull request closed without merging**